includes/query_processor.inc

```diff
@@ -333,6 +333,11 @@ class IslandoraSolrQueryProcessor {
     $object_results = array();
     if (isset($results)) {
       $solr_results = json_decode($results->getRawResponse(), TRUE);
+      if ($solr_results['response']['numFound'] === 0) {
+        // Invoke a hook for third-party modules to be notified that
+        // no results were found.
+        module_invoke_all('islandora_solr_query_no_results', $this);
```

Comment:
I think a general 'results returned' hook would be more appropriate. It might be worth noting that the results alter does not always fire, so some query returns would still be missed. If the desire is to only show when a front end 'islandora_search' has been executed the class may not be the right place in the code for the hook.